src/test/java/org/openrewrite/java/migrate/lang/UseTextBlocksTest.java

```diff
@@ -393,4 +392,78 @@ public static void main(String[] args) {
+     *                         https://www.springsource.org/spring-integration     \s
+     *                                                                             \s
+     *                     =========================================================\
+     *                     """;
+     *         String s3 = """
+     *                     \n=========================================================\
+     *                     \n                                                         \
+     *                     \n          Welcome to Spring Integration!                 \
+     *                     \n                                                         \
+     *                     \n    For more information please visit:                   \
+     *                     \n    https://www.springsource.org/spring-integration      \
+     *                     \n                                                         \
+     *                     \n=========================================================\
+     *                     """;
+     */
+    @Test
+    void newlineAtBeginningOfLines() {
+        rewriteRun(
+          version(
+            java(
+              """
+                class A {
+                    void welcome() {
+                        log("\\n========================================================="
+                            + "\\n                                                         "
+                            + "\\n          Welcome to Spring Integration!                 "
+                            + "\\n                                                         "
+                            + "\\n    For more information please visit:                   "
+                            + "\\n    https://www.springsource.org/spring-integration      "
+                            + "\\n                                                         "
+                            + "\\n=========================================================");
+                    }
+                    void log(String s) {}
+                }
+                """,
+              """
+                class A {
+                    void welcome() {
+                        log(\"""
+                           \s
+                            =========================================================
+                                                                                    \\s
+                                      Welcome to Spring Integration!                \\s
+                                                                                    \\s
+                                For more information please visit:                  \\s
+                                https://www.springsource.org/spring-integration     \\s
+                                                                                    \\s
+                            =========================================================\\
```

Comment:
@kunli2 Just out of interest: What is this last `\\` used for?

Reply:
Yeah, good question, it's to ensure that the closing delimiter is in a new line to manage indentation.
And this should be the best practice via https://docs.oracle.com/en/java/javase/20/text-blocks/index.html